src/promptflow-azure/promptflow/azure/_restclient/flow_service_caller.py

```diff
@@ -757,3 +762,98 @@ def init_workspace_cosmos(
             headers=self._get_headers(),
             **kwargs,
         )
+
+    @_request_wrapper()
+    def setup_workspace_cosmos(self, subscription_id, resource_group_name, workspace_name, body, **kwargs):
+        """Setup Cosmos DB for workspace/project."""
+        # TODO 3184158: use LROPoller for async API
+        # the standard way to call Azure async API is using LROPoller (Long Running Operations Poller)
+        # however, implement with naive way for now; will refine this part after Build
+        from azure.core.exceptions import (
+            ClientAuthenticationError,
+            ResourceNotFoundError,
+            map_error,
+        )
+        from promptflow.azure._restclient.flow.operations._trace_sessions_operations import (
+            _convert_request,
+            _models,
+            build_setup_trace_session_async_request,
+        )
+
+        headers = self._get_headers()
+        request_id = headers["x-ms-client-request-id"]
+        # below lines are copied and modified from TraceSessionsOperations.setup_trace_session_async
+        error_map = {401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError}
+        content_type = kwargs.pop("content_type", "application/json")
+        _json = self.caller.trace_sessions._serialize.body(body, "TraceDbSetupRequest")
+        request = build_setup_trace_session_async_request(
```

Comment:
why do we need to put such code in service caller?

Reply:
this API is an async API and we need to poll; the most standard way to do this is using `LROPoller` in `azure-core`, but it needs more time to implement. so copy some codes from generated rest client here to get async poll url from response header.